src/components/pages/Home.js

```diff
@@ -9,4 +17,15 @@ const Home = () => {
   );
 };
 
-export default Home;
+Home.propTypes = {
+  auth: PropTypes.object.isRequired
+};
+
+const mapStateToProps = state => ({
+  auth: state.auth
+});
+
+export default connect(
```

Comment:
Does this need to happen for every single components? Couldn't this be done at the top level and just make sure every top level components (under router) to get `auth` as their props? Similar to how you handled `PrivateRoute`, couldn't you make `PublicRoute` as well?